From: Avinash Dayanand <avinash.dayanand@intel.com>
Date: Wed, 27 Jul 2016 12:02:36 -0700
Subject: i40e: Force register writes to mitigate sync issues with iwarp VF
 driver
Patch-mainline: v4.9-rc1
Git-commit: 70df973b5eb48b19ba29105ecfecb3e50efe3c6d
References: bsc#1024346 FATE#321239 bsc#1024373 FATE#321247

This patch is a fix for the bug i.e. unable to create iwarp device
in VF. This is a sync issue and the iwarp device open is called even
before the PCI register writes are done.

Forcing the PCI register writes to happen just before it exits the
function.

Change-ID: I60c6a2c709da89e845f2764cc50ce8b7373c8c44
Signed-off-by: Avinash Dayanand <avinash.dayanand@intel.com>
Tested-by: Andrew Bowers <andrewx.bowers@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/intel/i40e/i40e_client.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/intel/i40e/i40e_client.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_client.c
@@ -812,7 +812,8 @@ static int i40e_client_setup_qvlist(stru
 			wr32(hw, I40E_PFINT_AEQCTL, reg);
 		}
 	}
-
+	/* Mitigate sync problems with iwarp VF driver */
+	i40e_flush(hw);
 	return 0;
 err:
 	kfree(ldev->qvlist_info);
